pkg/store/redis.go

```diff
@@ -22,6 +22,9 @@ type RedisStore struct {
 
 	// The maximum number of retry under race conditions.
 	MaxRetry int
+
+	// Redis connection stub for tests
+	mockGetConnection func() redis.Conn
```

Comment:
Removed mock method and extracted all logic to private methods that are tested now.